Homework1/BWT/Program.cs

```diff
@@ -0,0 +1,172 @@
+﻿namespace BWT;
+
+using System;
+using System.Text;
+
+class RotationsComparer : IComparer<int>
+{
+    private string str;
+
+    public RotationsComparer(string str)
+    {
+        this.str = str;
+    }
+
+    public int Compare(int first, int second)
+    {
+        for (var i = 0; i < str.Length; ++i)
+        {
+            if (str[(first + i) % str.Length] > str[(second + i) % str.Length])
+            {
+                return 1;
+            }
+            else if (str[(first + i) % str.Length] < str[(second + i) % str.Length])
+            {
+                return -1;
+            }
+        }
+        return 0;
+    }
+}
+
+class Program
+{
+    const int AlphabetSize = 65536;
+
+    //encodes the original string and returns BWT-encoded string
+    //and position of last original string's symbol 
+
+    static (string, int) Encoder(string encodingString)
+    {
+        var rotations = new int[encodingString.Length];
+
+        for (var i = 0; i < rotations.Length; ++i)
+        {
+            rotations[i] = i;
+        }
+
+        Array.Sort(rotations, new RotationsComparer(encodingString));
+
+        var result = new StringBuilder();
+        int positionOfStringEnd = 0;
+
+        for (var i = 0; i < encodingString.Length; ++i)
+        {
+            result.Append(encodingString[(rotations[i] + encodingString.Length - 1) % encodingString.Length]);
+
+            if (rotations[i] == 0)
+            {
+                positionOfStringEnd = i;
+            }
+        }
+
+        return (result.ToString(), positionOfStringEnd);
+    }
+
+    //decodes the BWT-encoded string with position of last original string's symbol
+    //and returns decoded original string
+    static string Decoder(string decodingString, int positionOfStringEnd)
+    {
+        StringBuilder resultString = new StringBuilder();
```

Comment:
```suggestion
        var resultString = new StringBuilder();
```
Эта переменная понадобится ой как не скоро. Перенеси её объявление ближе к месту её использования.
Так нагляднее код станет.